sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1134,3 +1139,110 @@ def convert_row_to_dict(self, row, schema):
       else:
         result[field.name] = self._convert_cell_value_to_dict(value, field)
     return result
+
+
+class BigQueryWriteFn(DoFn):
+  """A ``DoFn`` that streams writes to BigQuery once the table is created.
+  """
+
+  def __init__(self, table_id, dataset_id, project_id, batch_size, schema,
+               create_disposition, write_disposition, client):
+    self.table_id = table_id
+    self.dataset_id = dataset_id
+    self.project_id = project_id
+    self.schema = schema
+    self.client = client
+    self.create_disposition = create_disposition
+    self.write_disposition = write_disposition
+    self._rows_buffer = []
+    self._max_batch_size = batch_size or 500
```

Comment:
Why did you choose 500 for default batch size ? Please add a comment. (should we run some experiments to see if changing this has any affect ? )